src/revpi-functions

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+#
+# SPDX-License-Identifier: GPL-2.0
+#
+# Copyright: 2022 Kunbus GmbH
+#
+
+cm1-detection() {
+    grep -m 1 "model name" /proc/cpuinfo | grep -q v6l
+    return $?
```

Comment:
Absolutely. But what might be an error that is possible to occur? First of all, the requirement was that `/proc/cpuinfo` should be the source of information. And it might be that 1. the file does not exist or cannot be opened, or 2. that the format of cpuinfo has changed in a way that the first grep would find no match.

In both cases I'd consider the task impossible to solve. BTW: The first grep is only to make sure that the cpu type string is not taken out of context. I could remove it, if that is what you're suggesting. But that would introduce other errors.

Reply:
@zhan-kunbus  Just for clearing things up a bit, I've conducted following test:
```
$ grep -m 1 "stoppit" /proc/cpuinfo | grep -q v6l # no match at first grep
$ echo $?
1
$ grep -m 1 "stoppit" /proc/cpuinfo2 | grep -q v6l
grep: /proc/cpuinfo2: Datei oder Verzeichnis nicht gefunden
$ echo $?
1
```
So in any case of mentioned error above this would still be return 1 as an error, not zero.

Reply:
I just see that we have this problem, but no an absolut better solution. yes, agree, I think also there is no perfect solution here. but maybe it would be better to let the uncertainty in the future, in which case, if we have to deal with this issue

Reply:
maybe we can use this logic:  
if the string "model name      : ARMv7 Processor rev 3 (**v6l**)"  is contained in /proc/cpuinfo, the we see it as Core 1 and do the corresponding actions,  else (otherwise in any case) do the existed actions, which we see it as CM3, CM4S at moment, but maybe something different in the future.